Multirelay and LSD update for oc_relay.lsl

NikkiLacrima opened the pull request:
First version, tested with 4 simultaneous RLV sources and pingpong on relog.

Uses LSD to store restrictions per UUID
Uses LSD for mode and settings

NikkiLacrima committed:
Multirelay and LSD update for oc_relay.lsl

First version, tested with 4 simultaneous RLV sources and pingpong on relog.

Uses LSD to store restrictions per UUID
Uses LSD for mode and settings

NikkiLacrima committed:
Update oc_rlvsys.lsl

Removes handling of relay ping/pong protocol, moved entirely to oc_relay

NikkiLacrima committed:
Update oc_relay.lsl

Handle "Refuse" internally in relay, CMD_RELAY_SAFEWORD shall not trigger full safeword handling in rlvsys
Do we even need CMD_RELAY_SAFEWORD??
Handle Trusted Owners missing command
Stop relay listener when RLV_OFF

NikkiLacrima committed:
Update oc_relay.lsl

This is the correct source file

NikkiLacrima committed:
Update oc_rlvsys.lsl

CMD_RELAY_SAFEWORD shall not clear collar restrictions
Clean up "clear" handling and handle clear after g_iRlvActive set to 0 to allow clearing g_lRestrictions
removed CMD_ADDSRC and CMD_REMSRC (handled in relay)

NikkiLacrima committed:
Update oc_relay.lsl

Handle multiple pending ask sources
Code cleanup

NikkiLacrima commented:
Update Aug 25:
Handles multiple pending ask sources
Code cleanup

MinesweeperM commented:
Some trouble with capturing devices on relog.
On relog, relay returns av to device, where it was captured, then device trying to connect relay, but it does not answer, until main collar scripts load. So common devices (like AV_Sitter) decides that the relay is missing, and restrictions does not apply.

NikkiLacrima committed:
Update oc_relay.lsl

Some more fixes after testing

NikkiLacrima committed:
Merge branch 'main' into NikkiLacrima-Multirelay-and-LSD-patch-1

NikkiLacrima committed:
Merge branch 'main' into NikkiLacrima-Multirelay-and-LSD-patch-1

NikkiLacrima committed:
Move @detach handling to oc_rlvsys

NikkiLacrima committed:
Simplified startup, no reset on rez. Moved all @detach handling from oc_core

NikkiLacrima committed:
single state, fast startup improved memory handling

NikkiLacrima commented:
Changes 251018: Improved memory handling in relay, faster and simpler startup.

Hopefully this should be almost final version, still some debug output left.

NikkiLacrima committed:
Removed debug messages

Medea-Destiny reviewed (changes requested):
rlvsys and relay both still have script profiler active in menu (lines 132, 143 & 145 in rlvsys, 167, 177 and 179 in relay).  We might want to leave those in while we're putting this through beta, but should be commented out before merge.

NikkiLacrima committed:
Comment out memory profiling

NikkiLacrima committed:
Comment out memory profiling

NikkiLacrima committed:
minor fix in ReleaseAll()

When clearing all, the ping/pong target list should be cleared and not repopulated.

NikkiLacrima committed:
Use RLV_REFRESH after all settings read

Do not use RLV_ON when settings have been loaded, use RLV_REFRESH to signal a state update while RLV was already set ON.

NikkiLacrima committed:
Merge branch 'main' into NikkiLacrima-Multirelay-and-LSD-patch-1

NikkiLacrima committed:
Reverted previous

NikkiLacrima committed:
added g_iRlvActive flag to handle multiple RLV_ON

Medea-Destiny reviewed (approved):
Code looks in a really solid state to me, I can't think of anything more!

NikkiLacrima committed:
Added RLV_ON after "settings=sent"

NikkiLacrima committed:
Merge branch 'main' into NikkiLacrima-Multirelay-and-LSD-patch-1

NikkiLacrima committed:
Wearer access defaults to 1 if not set

also fix typo: relay_trustrust to relay_trusttrust

NikkiLacrima committed:
send RLV VERSION info after settings=sent
